Feat/search improve

JetDrag opened the pull request.

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 支持检索字段下钻

(cherry picked from commit 626d40a5779e94158afd4df15bf5610c685d7fd8)

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 支持查询收藏

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 支持统计图表

feat: 审计检索优化 --story=122841691

1. 添加用户通用配置接口

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 添加用户通用配置接口

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 添加用户通用配置接口

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 修正sql生成规则与配置

feat: 更新版本号至1.17.1-alpha.124

1. 支持统计图表

feat: 审计检索优化 --story=122841691

1. 支持统计图表

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 变更接口保持和规则审计一致

0RAJA commented on src/backend/services/web/query/resources/doris.py:
```diff
@@ -70,30 +83,44 @@ def build_sql(self, validated_request_data):
+        """
+        构建日志查询SQL，返回所有数据
+        """
+        sql_builder = super().build_sql(validated_request_data)
```
这里可以直接调用self.get_sql_builder()而不用重写了

0RAJA commented on src/backend/services/web/query/resources/doris.py:
```diff
@@ -115,14 +142,74 @@ def perform_request(self, validated_request_data):
+    name = gettext_lazy("日志统计统计")
+    RequestSerializer = CollectorSearchStatisticReqSerializer
+    serializer_class = CollectorSearchStatisticRespSerializer
     audit_action = ActionEnum.SEARCH_REGULAR_EVENT
```
这里是否应该是：REGULAR_EVENT_STATISTIC

0RAJA commented on src/backend/services/web/query/resources/doris.py:
```diff
@@ -115,14 +142,74 @@ def perform_request(self, validated_request_data):
+    name = gettext_lazy("日志统计查询(All)")
+    RequestSerializer = CollectorSearchAllStatisticReqSerializer
+    ResponseSerializer = CollectorSearchStatisticRespSerializer
+    audit_action = ActionEnum.REGULAR_EVENT_STATISTIC
```
这里增加后内部直接调用该 resource 的请求也会上报审计日志，看看是否需要呢

0RAJA commented on src/backend/services/web/query/utils/doris.py:
```diff
@@ -120,3 +134,84 @@ def build_count_sql(self) -> str:
+        """
+        return super()._build_base_table(table).as_("event_table")
+
+    def build_statistic_sql(self, field_name: str, field_type: FieldType) -> dict[str, str]:
```
逻辑清晰，写的非常好

0RAJA commented on src/backend/services/web/query/models.py:
```diff
@@ -0,0 +1,21 @@
+from core.models import OperateRecordModel
+
+
+class FavoriteSearch(OperateRecordModel):
```
这个是否和GeneralConfig重复了呢,输入参数比较多变，统一存储为 json 是否更灵活呢

JetDrag replied:
去掉了

0RAJA commented on src/backend/services/web/query/serializers.py:
```diff
@@ -222,15 +224,24 @@ class QuerySearchResponseSerializer(serializers.Serializer):
 
-    field_name = serializers.ChoiceField(label="字段名", choices=COLLECT_SEARCH_CONFIG.allowed_query_field_choices)
+    raw_name = serializers.CharField(label=gettext_lazy("Raw Name"))
+    field_type = serializers.ChoiceField(label=gettext_lazy("Field Type"), choices=FieldType.choices, default=None)
```
allow_null = True

0RAJA commented on src/backend/services/web/query/serializers.py:
```diff
@@ -356,7 +375,7 @@ def _build_system_conditions(cls, validated_request_data) -> List[dict]:
             return [
                 {
-                    "field_name": SYSTEM_ID.field_name,
+                    "field": {"raw_name": SYSTEM_ID.field_name, "field_type": "string", "keys": []},
```
field_type: FieldType.STRING.value

0RAJA commented on src/backend/apps/meta/utils/fields.py:
```diff
@@ -343,6 +346,24 @@
+    property={
+        "dynamic_content": False,
+        "sub_keys": [
+            ["action_id"],
```
这里的协议针对以下场景是否可以满足呢：
1. 多级嵌套的情况呢，例如 k1,k2和k1,k3，当前协议为：[["k1","k2"],["k1","k3"]]，需要重复多次
2. 嵌套子字段需要提供别名以及字段类型

是否需要修改为类似的形式：
```json
[
    {
        "field_name": "k1",
        "alias_name": "k1",
        "field_type": "string",
        "sub_keys": [
            {
                "field_name": "k2",
                "alias_name": "k2",
                "field_type": "string"
            },
            {
                "field_name": "k3",
                "alias_name": "k2",
                "field_type": "string"
            }
        ]
    }
]
```

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 优化字段统计工具返回

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. 变更standard field sub_key类型为dict。
2. standard field property添加type hint
3. 一些代码优化。

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. sub key添加field_alias。

JetDrag committed:
feat: 审计检索优化 --story=122841691

1. cr修正。
